### PR TITLE
Disable the flag check on method without stackmaps

### PR DESCRIPTION
**Pull request by ChengJin01**
The change is to disable the check on the uninitializedThis
flag on old version classfiles (< 51) as such check is only
applied to classfiles with stackmaps which was introduced
since Java 7(class version is 51).

Fix: #5676

Signed-off-by: Cheng Jin <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
Reviewer: @DanHeidinga 

FYI: @pshipton

**Comment by DanHeidinga** [comment]
Jenkins test sanity,extended xlinux jdk8

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test extended xlinux jdk8

File: runtime/bcverify/rtverify.c [hunk 1]
```diff
@@ -307,7 +307,7 @@ matchStack(J9BytecodeVerificationData * verifyData, J9BranchTargetStack *liveSta
 
 	/* Jazz103: 120689 */
 	/* Target stack frame flag needs to be subset of ours. See JVM sepc 4.10.1.4 */
-	if (liveStack->uninitializedThis && !targetStack->uninitializedThis) {
+	if (!verifyData->createdStackMap && liveStack->uninitializedThis && !targetStack->uninitializedThis) {
```
Comment (DanHeidinga) [inline runtime/bcverify/rtverify.c]: This should have a comment in the code explaining why it's not applicable to the createdStackMap case.

The code may end up looking cleaner if written in this form once the comment is added:
```
/* Long comment describing
 * why this check isn't applicable to
 * createdstackMaps
 */
if (!verifyData->createdStackMap) {
    /* Target stack frame flag needs to be subset of ours. See JVM sepc 4.10.1.4 */
    if (liveStack->uninitializedThis && !targetStack->uninitializedThis) {
        .....
    }
}
```

Comment (ChengJin01) [reply runtime/bcverify/rtverify.c]: Already updated the code & comment against the suggestion above.

**Comment by DanHeidinga** [comment]
Jenkins test extended xlinux jdk8

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by DanHeidinga** [comment]
@ChengJin01 Can you cherrypick this commit into a PR against the `v0.14.2-release` release branch?

**Comment by ChengJin01** [comment]
Already created the PR against `v0.14.2-release`.

**Comment by DanHeidinga** [comment]
PR for the 0.14.2 release: https://github.com/eclipse/openj9/pull/5740

**Pull request merged**